tpm_quote/src/lib.rs

```diff
@@ -84,7 +83,44 @@ pub fn verify_quote(
         }
     }
 
-    Ok(quote_info.to_owned())
+    verify_quote_policy(&attestation_data, policy)?;
+
+    let quote_info = match attestation_data.attested() {
+        AttestInfo::Quote { info } => Ok(info),
+        _ => Err(Error::QuoteError("unexpected attestion type".to_owned())),
+    }?;
```

Comment:
you can use the form `let AttestInfo::Quote { info } = attestation_data.attested() else { return Err(...) };`